chapters/planning-data-work.tex

```diff
@@ -151,162 +469,185 @@ \subsection{Code management}
-
-Breaking your code into readable steps is also good practice on code organization.
+Finally, use the header to track the inputs and outputs of the script.
+When you are trying to track down which code creates which data set, this will be very helpful.
+While there are other ways to document decisions related to creating code
+(GitHub offers a lot of different documentation options, for example),
+the information that is relevant to understand the code should always be written in the code file.
+
+In the script, alongside the code, are two types of comments that should be included.
+The first type of comment describes what is being done.
+This might be easy to understand from the code itself
+if you know the language well enough and the code is clear,
+but often it is still a great deal of work to reverse-engineer the code's intent.
+Writing the task in plain English (or whichever language you communicate with your team on)
+will make it easier for everyone to read and understand the code's purpose
+-- and also for you to think about your code as you write it.
+The second type of comment explains why the code is performing a task in a particular way.
+As you are writing code, you are making a series of decisions that
+(hopefully) make perfect sense to you at the time.
+These are often highly specialized and may exploit a functionality
+that is not obvious or has not been seen by others before.
+Even you will probably not remember the exact choices that were made in a couple of weeks.
+Therefore, you must document your precise processes in your code.
+
+Code organization means keeping each piece of code in an easily findable location.
+  \index{code organization}
+Breaking your code into independently readable ``chunks'' is one good practice on code organization,
+because it ensures each component does not depend on a complex program state
+created by other chunks that are not obvious from the immediate context.
 One way to do this is to create sections where a specific task is completed.
 So, for example, if you want to find the line in your code where a variable was created,
 you can go straight to \texttt{PART 2: Create new variables},
-instead of reading line by line of the code.
-RStudio makes it very easy to create sections, and compiles them into an interactive script index.
+instead of reading line by line through the entire code.
+RStudio, for example, makes it very easy to create sections,
+and it compiles them into an interactive script index for you.
 In Stata, you can use comments to create section headers,
-though they're just there to make the reading easier.
-Adding a code index to the header by copying and pasting section titles is the easiest way to create a code map.
-You can then add and navigate through them using the find command.
+though they're just there to make the reading easier and don't have functionality.
+Adding an index to the header by copying and pasting section titles is the easiest way to create a code map.
```

Comment:
we do explain code map anywhere. Should we call it an index or a table of content?

Reply:
80a42c84df22acb662032c394e36305777a479b0